// @coinbase/x402/dist/esm/index.mjs
// src/index.ts
import { generateJwt } from "@coinbase/cdp-sdk/auth";
var COINBASE_FACILITATOR_BASE_URL = "https://api.cdp.coinbase.com";
var COINBASE_FACILITATOR_V2_ROUTE = "/platform/v2/x402";
var X402_SDK_VERSION = "0.5.1";
var CDP_SDK_VERSION = "1.29.0";
async function createAuthHeader(apiKeyId, apiKeySecret, requestMethod, requestHost, requestPath) {
  const jwt = await generateJwt({
    apiKeyId,
    apiKeySecret,
    requestMethod,
    requestHost,
    requestPath
  });
  return `Bearer ${jwt}`;
}
function createCorrelationHeader() {
  const data = {
    sdk_version: CDP_SDK_VERSION,
    sdk_language: "typescript",
    source: "x402",
    source_version: X402_SDK_VERSION
  };
  return Object.keys(data).map((key) => `${key}=${encodeURIComponent(data[key])}`).join(",");
}
function createCdpAuthHeaders(apiKeyId, apiKeySecret) {
  const requestHost = COINBASE_FACILITATOR_BASE_URL.replace("https://", "");
  return async () => {
    apiKeyId = apiKeyId ?? process.env.CDP_API_KEY_ID;
    apiKeySecret = apiKeySecret ?? process.env.CDP_API_KEY_SECRET;
    if (!apiKeyId || !apiKeySecret) {
      throw new Error(
        "Missing environment variables: CDP_API_KEY_ID and CDP_API_KEY_SECRET must be set when using default facilitator"
      );
    }
    return {
      verify: {
        Authorization: await createAuthHeader(
          apiKeyId,
          apiKeySecret,
          "POST",
          requestHost,
          `${COINBASE_FACILITATOR_V2_ROUTE}/verify`
        ),
        "Correlation-Context": createCorrelationHeader()
      },
      settle: {
        Authorization: await createAuthHeader(
          apiKeyId,
          apiKeySecret,
          "POST",
          requestHost,
          `${COINBASE_FACILITATOR_V2_ROUTE}/settle`
        ),
        "Correlation-Context": createCorrelationHeader()
      },
      list: {
        Authorization: await createAuthHeader(
          apiKeyId,
          apiKeySecret,
          "GET",
          requestHost,
          `${COINBASE_FACILITATOR_V2_ROUTE}/discovery/resources`
        ),
        "Correlation-Context": createCorrelationHeader()
      }
    };
  };
}
function createFacilitatorConfig(apiKeyId, apiKeySecret) {
  return {
    url: `${COINBASE_FACILITATOR_BASE_URL}${COINBASE_FACILITATOR_V2_ROUTE}`,
    createAuthHeaders: createCdpAuthHeaders(apiKeyId, apiKeySecret)
  };
}
var facilitator = createFacilitatorConfig();
export {
  createAuthHeader,
  createCdpAuthHeaders,
  createCorrelationHeader,
  createFacilitatorConfig,
  facilitator
};
